CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+
+  * **(Option A)** When continuing a chained method invocation on
+    another line keep the `.` on the second line.
+
+    ```Ruby
+    # bad - need to consult first line to understand second line
+    one.two.three.
+      four
+
+    # good - it's immediately clear what's going on the second line
+    one.two.three
+      .four
+    ```
+
+  * **(Option B)** When continuing a chained method invocation on another line,
+    include the `.` on the first line to indicate that the
+    expression continues.
+
+    ```Ruby
+    # bad - need to read ahead to the second line to know that the chain continues
+    one.two.three
+      .four
+
+    # good - it's immediately clear that the expression continues beyond the first line
+    one.two.three.
+      four
+    ```
+
+  A discussion on the merits of both alternative styles can be found
+  [here](https://github.com/bbatsov/ruby-style-guide/pull/176).
+
+* Align the parameters of a method call if they span more than one line. When aligning parameters is not appropriate due to line-length constraints, single indent for the lines after the first is also acceptable.
+
+  ```Ruby
+  # starting point (line is too long)
+  def send_mail(source)
+    Mailer.deliver(to: 'bob@example.com', from: 'us@example.com', subject: 'Important message', body: source.text)
+  end
+
+  # bad (double indent)
+  def send_mail(source)
+    Mailer.deliver(
+        to: 'bob@example.com',
+        from: 'us@example.com',
+        subject: 'Important message',
+        body: source.text)
+  end
+
+  # good
+  def send_mail(source)
+    Mailer.deliver(to: 'bob@example.com',
+                   from: 'us@example.com',
+                   subject: 'Important message',
+                   body: source.text)
+  end
+
+  # good (normal indent)
+  def send_mail(source)
+    Mailer.deliver(
+      to: 'bob@example.com',
+      from: 'us@example.com',
+      subject: 'Important message',
+      body: source.text
+    )
+  end
+  ```
+
+* Align the elements of array literals spanning multiple lines.
+
+  ```Ruby
+  # bad - single indent
+  menu_item = ['Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam',
+    'Baked beans', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam']
+
+  # good
+  menu_item = [
+    'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam',
+    'Baked beans', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam'
+  ]
+
+  # good
+  menu_item =
+    ['Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam',
+     'Baked beans', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam']
+  ```
+
+* Add underscores to large numeric literals to improve their readability.
+
+  ```Ruby
+  # bad - how many 0s are there?
+  num = 1000000
+
+  # good - much easier to parse for the human brain
+  num = 1_000_000
+  ```
+
+ * Use RDoc and its conventions for API documentation.  Don't put an empty line between the comment block and the `def`.
+
+  * Limit lines to 80 characters.
+  * Avoid trailing whitespace.
```

Comment:
The only reason 'avoid trailing whitespace' is a good idea is because it seems randomly folks editors have this obsessive-configuration that will strip _all_ trailing whitespace when they write to files. This causes unintentional whitespace changes to litter contributors' pull requests and causes confusion.

To avoid this confusion during PR review and to avoid accidental changes, maybe we enforce no trailing whitespace?

Otherwise, trailing whitespace has literally zero impact on the actual code or its readability, so I'm fine if we don't enforce it.